mantidproject/mantid
# Engg:: Add additional tests for invalid focused files

Description of work.
Moved `splitFittingDir( )` out of the view into the presenter as it doesn't need to interact with the GUI and changed it to always split the input (previously it only split if "ENGINX_" was in the filename which restricts the instruments we can use on this interface)
Additionally updated a catch block to catch a bad cast if we have unexpected data in the file. 

The code changes will be refactored as part of #17301 - this is only functional changes. 

**To test:**
Open Interfaces -> Diffraction -> Engineering Diffraction. 
Using a plain EnginX file such as `ENGINX00193749` in System test data (i.e. one that is not fitted) try to open the file in the fitting tab and run the fit

Mantid should not crash and should stop the fitting algorithm notifying the user that they must use a focused file. 

Fixes #17064 .

Does not need to be in the release notes.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
